app/code/Magento/NewRelicReporting/Plugin/StatePlugin.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\NewRelicReporting\Plugin;
+
+use Magento\Framework\App\State;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\NewRelicReporting\Model\Config;
+use Magento\NewRelicReporting\Model\NewRelicWrapper;
+
+class StatePlugin
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var NewRelicWrapper
+     */
+    private $newRelicWrapper;
+
+    /**
+     * @param Config $config
+     * @param NewRelicWrapper $newRelicWrapper
+     */
+    public function __construct(
+        Config $config,
+        NewRelicWrapper $newRelicWrapper
+    ) {
+        $this->config = $config;
+        $this->newRelicWrapper = $newRelicWrapper;
+    }
+
+    /**
+     * Set separate appname
+     *
+     * @param State $subject
+     * @param null $result
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterSetAreaCode(State $state, $result)
+    {
+        if (!$this->shouldSetAppName()) {
+            return;
+        }
+
+        try {
+            $this->newRelicWrapper->setAppName($this->appName($state));
+        } catch (LocalizedException $e) {
+            return;
```

Comment:
1. Why should we ignore such exception?
2. Probably we should have `return $result` here